### PR TITLE
mps fix

### PR DESCRIPTION
**Pull request by Jerry-Master**
By adding that single conversion from float64 to float32 the integrated gradients is fully compatible with MPS backend, which it wasn't previously. The change can be accepted as valid for any backend since float64 is a precision almost nobody using nowadays in deep learning. It seems that the default for torch.tensor is not updated with current trends.

### CODE REVIEW

**Review by vivekmig — APPROVED** [review]
Thanks so much for this fix @Jerry-Master !
